opensearch-operator/pkg/helpers/helpers.go

```diff
@@ -359,3 +360,28 @@ func CompareVersions(v1 string, v2 string) bool {
 	ver2, _ := version.NewVersion(v2)
 	return err == nil && ver1.LessThan(ver2)
 }
+
+func GetJvmHeapSize(nodePool *opsterv1.NodePool) string {
+	jvmHeapSizeTemplate := "-Xmx%s -Xms%s"
+
+	if nodePool.Jvm == "" {
+		memoryLimit := nodePool.Resources.Requests.Memory()
+		nodePoolMemorySize, _ := resource.ParseQuantity(memoryLimit.String())
+
+		// Memory request is not present
+		if nodePoolMemorySize.IsZero() {
+			maximumJavaHeapSize, initialJavaHeapSize := "512M", "512M"
+			return fmt.Sprintf(jvmHeapSizeTemplate, maximumJavaHeapSize, initialJavaHeapSize)
+		}
+
+		// Java heap size is set to half of the memory request
+		memoryRequestInt64, _ := nodePoolMemorySize.AsInt64()
+		memoryRequestQty := resource.NewQuantity(memoryRequestInt64/2, resource.BinarySI)
+		megabytes := float64(memoryRequestQty.Value() / 1024.0 / 1024.0)
+
+		maximumJavaHeapSize, initialJavaHeapSize := fmt.Sprintf("%vM", megabytes), fmt.Sprintf("%vM", megabytes)
+		return fmt.Sprintf(jvmHeapSizeTemplate, maximumJavaHeapSize, initialJavaHeapSize)
```

Comment:
No need to use two variables. Just use one and provide it twice to the Sprintf call.

Reply:
Yes, maximumJavaHeapSize and initialJavaHeapSize are combined in `heapSize` variable know.